job/tests/test_views.py

```diff
@@ -0,0 +1,230 @@
+            "name": "برنامه نویس Go",
+            "job_slug": slug_helper("برنامه نویس Go"),
+            "cover": "/default/system/cover2.jpg",
+            "icon": "fas fa-home2",
+            "description": "برنامه نویس Go",
+            "is_deleted": True,
+            "approved": True,
+        }
+        self.client.credentials(HTTP_AUTHORIZATION="JWT " + self.token)
+        response = self.client.put(url, data, format="json")
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response._headers["content-type"][1], "application/json")
+        # Constant fields
+        self.assertEqual(job.id, response_data["data"]["id"])
+        self.assertEqual(job.name, response_data["data"]["name"])
+        self.assertEqual(job.job_slug, response_data["data"]["job_slug"])
+        self.assertFalse(job.is_deleted)
+        self.assertFalse(job.approved)
+        # Changed fields
+        self.assertEqual(response_data["data"]["cover"], data["cover"])
+        self.assertEqual(response_data["data"]["icon"], data["icon"])
+        self.assertEqual(response_data["data"]["description"], data["description"])
+
+        ## Updating non-existent object
+        url = "/job/5000/update/"
+        response = self.client.put(url, data, format="json")
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(response_data["message"], "Instance does not exist.")
+
+    def test_job_update_patch(self):
+        ## Updating existent object
+        job = Job.objects.first()
+        url = f"/job/{job.id}/update/"
+        partial_data = {
+            "cover": "/default/system/cover3.jpg",
+            "icon": "fas fa-home3",
+            "description": "برنامه نویس Rust",
+        }
+
+        data = {
+            "id": job.id,
+            "name": "برنامه نویس وب",
+            "job_slug": "برنامه-نویس-وب",
+            "cover": "/default/system/cover3.jpg",
+            "icon": "fas fa-home3",
+            "description": "برنامه نویس Rust",
+            "is_deleted": False,
+            "approved": False,
+        }
+        self.client.credentials(HTTP_AUTHORIZATION="JWT " + self.token)
+        response = self.client.patch(url, partial_data, format="json")
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response._headers["content-type"][1], "application/json")
+        self.assertEquals(response_data["data"], data)
+
+        ## Updating non-existent object
+        url = "/job/5000/update/"
+        response = self.client.patch(url, partial_data, format="json")
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(response_data["message"], "Instance does not exist.")
+
+    def test_job_delete(self):
```

Comment:
you could divide this test and others to multi tests for example test_job_delete_view_when_exists & test_job_delete_view_when_not_exists try this on other tests too.